infrastructure/docker-compose.prod-deploy.yml

```diff
@@ -46,9 +46,52 @@ services:
       - NODE_ENV=production
 
   scheduler:
-      environment:
+    environment:
       - NODE_ENV=production
-      
+
   documents:
     environment:
       - NODE_ENV=production
+
+  wg-easy:
```

Comment:
I think we need to move this to docker-compose-vpn.deploy.yml.  Then deploy.sh script should pick that up only if we are provisioning a wireguard VPN

Reply:
I was thinking something similar in the [compose file PR](https://github.com/opencrvs/opencrvs-farajaland/pull/808) for making the MongoDB clusters dynamic with a `docker-compose.cluster.yml`. In the end, as per our conversation yesterday, decided to configure the deployment sizes and replicas per environment in the environment files to only reflect how the Farajaland setup is. 

I agree, VPN should be a default for production environments. Doing it like we're doing it here, so deploying it to a production machine, is not the right choice for real production environments and the VPN configuration will be different for different network setups. 

**If we want to make this generic, we would at least need to:**
- Ensure production machine is in a private internal network
- Provide a "jump host" mechanism for provisioning and deployment and other pipelines. This is an SSH proxy setting so you can connect to machine B in internal network by first connection to A in the public internet.
- DNS server for clients connecting with a domain. This is because if the application server doesn't have a public IP address, there's nothing to attach the `register.crvs.fr` domain to. The DNS server needs to redirect the client from `register.crvs.fr` to the internal address e.g. `192.168.1.4`.
- Make it configurable to which internal network(s) the Wireguard server attaches the connecting client
- Make the Wireguard server be setup by provisioning scripts and not be a docker container so it can access the network interfaces of the host machines to make the previous point possible

In the long term, I'd also like to phasing out deploy.sh completely, bringing more logic into the compose files similarly to what Kubernetes + Helm combination would be.

Reply:
Lets move this to the staging compose file